File: backend-server/src/main/kotlin/models/Token.kt
package com.ite393group5.models
import com.ite393group5.utilities.DateSerializer
import kotlinx.serialization.Serializable
import java.time.LocalDate


@Serializable
data class Token(
    val bearerToken: String,
    val refreshToken: String,
    @Serializable(with = DateSerializer::class)
    val expirationTimeDate: LocalDate?
)

File: web/includes/models/Token.php
<?php

namespace models;

use DateTime;
use Exception;

class Token
{
    public $bearerToken;
    public $refreshToken;
    public $expirationTimeDate; // Will be stored as "YYYY-MM-DD"

    public function __construct($bearerToken,$refreshToken,$expirationTimeDate) {
        $this->bearerToken = $bearerToken;
        $this->refreshToken = $refreshToken;
        $this->expirationTimeDate = $expirationTimeDate;
    }

    // Convert JSON string to Token object
    public static function fromJson($json) {
        try {
            $data = json_decode($json, true); // Decode JSON to array
            if (!$data) {
                throw new Exception("Invalid JSON");
            }

            return new Token(
                $data['bearerToken'],
                $data['refreshToken'],
                $data['expirationTimeDate']
            );
        } catch (Exception $e) {
            return null;
        }
    }

    // Convert Token object to JSON
    public function toJson() {
        return json_encode(get_object_vars($this));
    }
}


